### ISSUE TITLE
Settings-based confirmation for using @all notification on messages

### ISSUE THREAD

**Issue by gmsecrieru** [RocketChat/Rocket.Chat#909]
Even though we had a few funny times when someone shoot an `@all` notification in the middle at the night on the Demo Server, what do you folks think about the idea of adding a settings-based confirmation before using it? 

![screen shot 2015-01-30 at 3 36 44 pm](https://cloud.githubusercontent.com/assets/190883/10118887/3af3d80c-645b-11e5-811d-0943a9813425.png)

Maybe it could be related to rules such as the number of users in the channel and/or number of different timezones etc.

**Comment by geekgonecrazy** [comment]
Referencing #1268 as has bounty.

**Comment by engelgabriel** [comment]
Refs https://github.com/RocketChat/Rocket.Chat/issues/1654

**Comment by joshuahaber** [comment]
Is it possible to disable the @all functionality completely?  If someone tries to send to @all, it doesn't do anything?

**Comment by MartinSchoeler** [comment]
@johanekhager you can remove the `mention-all` permission from all users

**Comment by FaizanZahid** [comment]
+1

**Comment by rodrigok** [comment]
We already have options to restrict the usage of `@all` in large rooms.

This issue is more related to an UX improvements with some intelligence advising users when some of the ones that will be notified aren't in work hours.

So I set this as a GSoC idea.

**Comment by Saibaba1009** [comment]
@rodrigok hey is this issue is still to be solved can I work on this and do necessary improvements.

**Comment by atpug22** [comment]
@rodrigok @engelgabriel  is this issue still unsolved then can I work on this??

**Comment by rodrigok** [comment]
@atpug22 yes, we didn't solve this yet, feel free to propose a solution :)

**Comment by AswinCG2002** [comment]
Hello everyone, can someone tell me what changes are you expecting on this issue. Is this issue specific to UI/UX designers?

**Comment by micksharma** [comment]
Can I work on this bug ?

**Comment by Akshanshkaushal** [comment]
Hello, @gmsecrieru , @tassoevan and @rodrigok.

I would like to work on this problem.
Let me suggest a fix.
When @all is triggered, a confirmation prompt message will be shown based on the number of users and users in specific timezones. 

Along with this, there is an option to schedule messages based on timezones. The backend will fetch the different timezones based on who belongs to that group and group people based on timezones. Now users can send messages based on timezones. 

for instance, there are 18 people in a group . 9 from India, 3 from Germany, and the rest from USA. So when all is triggered, along with confirmation, there is an option to schedule messages for these 3 different timezones. Based on scheduled messages, they are delivered to people belonging to that timezone. 


The solution involves two key enhancements to the notification system:

-->**Confirmation Prompt Based on Users and Timezones:**
When the @all notification is triggered, a confirmation prompt will appear.
This prompt will be shown based on the number of users and their respective timezones.

-->**Scheduled Messages Based on Timezones:
1: Provide an option to schedule messages according to the users' timezones.
The backend will determine the different timezones of users in the group.
Users can then send scheduled messages specific to each time zone.

**Comment by DEEP13-2-5** [comment]
Hi maintainers 👋  

I’ve read through the issue and the previous discussion.  
From my understanding, this would involve showing a confirmation (modal / popup) when a user tries to use `@all`, potentially controlled via settings or permissions.

I’d like to work on this issue if it’s still available.  
Before starting, I wanted to confirm:
- Should this be only a client-side confirmation, or also configurable via server settings?
- Are there existing confirmation dialog patterns/components in Rocket.Chat that I should reuse?

Happy to wait for guidance before proceeding. Thanks!

**Comment by akankshahu** [comment]
Hi @rodrigok @matheus.cardoso @Sing-Li,

I am Akanksha, a GSoC 2026 applicant. I’ve set up the environment on Arch Linux and would like to propose a modern fix for this:I am ready to submit a Draft PR for this flow. May I be assigned?

**Comment by Rohit544** [comment]
Hi, I’m Naveen. I attended the Rocket.Chat community workshop and would like to work on this issue as my first contribution.

Could I be assigned to it?

**Comment by Ayush725606** [comment]
Hi, I'm a GSoC 2026 applicant and I'd like to work on this issue. Can a mentor guide me on where to start?

**Comment by Ayush725606** [comment]
Hi, I’ve submitted a small PR improving code readability in the DocumentationCard component.

While exploring mentions, I noticed that @all can sometimes appear as @All due to formatting. 

So I’m planning to handle detection using a case-insensitive approach (e.g., regex /@all/i) to ensure reliability.

My approach is:
- Detect @all in MessageComposer (case-insensitive)
- Intercept message send flow
- Add confirmation modal using Fuselage UI
- Add backend validation

I’d appreciate guidance on the best place to hook into the message send flow.

Thanks!

**Comment by Kaashya** [comment]
Hi, I am a GSoc 2026 and would like to work on this issue. Could you please confirm if it’s available?

**Comment by namann5** [comment]
Hi, I'm a GSoC 2026 applicant. I'd like to work on this — can someone guide me on the expected behavior and assign this to me?

**Comment by rhodesgarret56-cell** [comment]
Thanks for the interest everyone.

Before assigning this, it would be good if one of you could take a first
pass at exploring the code and outlining a concrete approach (similar to
what @Ayush725 started). That makes it easier to give targeted feedback and
avoid duplicated work.

Happy to guide once there’s a bit more detail on the proposed
implementation.

**Comment by KCreations92** [comment]
Hello Everyone! I'm Kashif, a GSoC 2026 applicant interested in the AI Rocket.Chat Apps Generator project. I'd love to contribute to this issue as part of my community engagement. Could a maintainer guide me on the best approach to tackle this?  Thanks!

**Comment by adityasingh6533** [comment]
Hi, I'm a GSoC 2026 applicant. I'd like to work on this — can someone guide me on the expected behavior and assign this to me?

**Comment by eshaanag** [comment]
Hi team, I would love to take this on. 

My plan is to add a new `Message_Confirm_All_Here_Mentions` boolean to the admin settings, and intercept the dispatch inside the client `sendMessage` flow to trigger a `@rocket.chat/fuselage` confirmation modal before proceeding. 

Could you assign this issue to me?

**Comment by Adarsh-Me** [comment]
Hi! I'd like to work on this issue — adding a confirmation setting before @all notifications go out. I've read through the relevant settings code and the linked UI discussion. Could you please assign it to me? Thanks!
